src/samVisualize/rendering/icons.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { readdirSync } from 'fs'
+import { join } from 'path'
+import * as vscode from 'vscode'
+/**
+ * Maps a resource type to a Webview URI pointing to a corresponding svg icon
+ */
+export type IconURIMap = {
+    [index: string]: string
+}
+/**
+ * Creates a map between a resource type and a webview URI pointing to a corresponding svg icon
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html
+ * @param iconDir The path to the directory holding icons
+ * @param webview The webview from which URIs pointing to svg icons is produced
+ * @returns an IconURIMap mapping each svg icon in `iconDir`
+ */
+export function generateIconsMap(iconDir: string, webview: vscode.Webview): IconURIMap {
```

Comment:
Where is this used? Can't find it in this PR.

Reply:
It's used by the extension when a template is rendered, and the output is passed to the rendering function

Reply:
Is it used in this PR?

Reply:
Outside of tests, not in this PR